File: android-app/app/src/main/java/com/anonymous/chat/models/PatchNote.java
package com.anonymous.chat.models;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

public class PatchNote implements Serializable {
    private String version;
    private String title;
    private String date;
    private List<String> items = new ArrayList<>();

    public PatchNote() {}

    public String getVersion() { return version != null ? version : ""; }
    public void setVersion(String version) { this.version = version; }

    public String getTitle() { return title != null ? title : ""; }
    public void setTitle(String title) { this.title = title; }

    public String getDate() { return date != null ? date : ""; }
    public void setDate(String date) { this.date = date; }

    public List<String> getItems() {
        if (items == null) items = new ArrayList<>();
        return items;
    }
    public void setItems(List<String> items) { this.items = items; }
}

File: ios-app/AnonymousChat/Models/PatchNote.swift
import Foundation

public struct PatchNote: Codable, Identifiable {
    public var id: String { version }
    public let version: String
    public let date: String
    public let title: String
    public let items: [String]

    public init(version: String, date: String, title: String, items: [String]) {
        self.version = version
        self.date = date
        self.title = title
        self.items = items
    }
}
